### PR TITLE
New VReplication workflows cli UX. Allow reads/writes to be switched independently

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
The motivation and design for this issue are outlined in the RFC at #7225. Some implementation details are provided below: more detailed documentation will be added to the vitess.io website soon.

The main change is that all subcommands for a MoveTables or Reshard workflow is routed through the new VReplicationWorkflow object. VReplicationWorkflow  first deduces the state of a workflow from the parameters in the _vt sidecar and topo and invokes the associated wrangler methods. The only changes made in wrangler related to this PR is to allow switching tablet types in any order. 

## vtctld mods

MoveTables and Reshard commands now take a -v2 parameter which causes the new functionality to be invoked. These will continue to run side-by-side with the current functionality. We will first deprecate v1 and remove v1-only code in a subsequent release making v2 functionality as default.

## Routing rule changes

MoveTables v1 uses routing rules to tell vtgate, which keyspace to route queries to. In v1 we always need to run SwitchReads before SwitchWrites. MoveTables sets up routing rules pointing primary/replica/rdonly tables to the source keyspace. SwitchReads would change the routing rules for related tablet types to point to the target and reversal point it back to the source. SwitchWrites would delete all rules except the one for the table. 

Since tablet types can now be switched (and reversed) in any order we keep the routing rules through the lifetime of the workflow. We keep the routing rules in place and currently DropSources does not delete them. This PR continues to keep the routing rules. A follow-up PR will add the removal of routing rules to the Complete/Abort subcommands.

## Abort

It is possible to abort and cleanup a created workflow as long as traffic has not been switched. This deletes the vreplication artifacts and optionally, delete data: target tables/shards.

## Progress

We report the progress of a workflow by showing the percentage of data copied across targets, if workflow is in copy state, and the lag between the last source event after the copy phase is completed. It is too expensive to get actual row counts of tables, so we use the statistics available in information_schema to approximate copy progress. This data can be significantly off (upto 50-60%) depending on the utilization of the underlying mysql server resources. 

Notes:

* We are still marking the streams as FROZEN when we switch primary traffic even though replica and rdonly traffic may not be switched yet. This will be addressed as part of a major refactor of the vreplication related tables in the sidecar _vt table.

##  Todos: upcoming PRs will achieve these
* delete routing rules, update vschema after move tables #7234
* Add a check to the Start command to monitor that the command actually starts and report any errors or lack of progress
* dry-run

### CODE REVIEW

**Review by sougou — COMMENTED** [review]
Looking good so far. Let's keep going.

File: go/test/endtoend/vreplication/cluster.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 )
 
 var (
-	debug              = false // set to true to always use local env vtdataroot for local debugging
+	debug              = true // set to true to always use local env vtdataroot for local debugging
```
Comment (sougou) [inline go/test/endtoend/vreplication/cluster.go:24]: Probably need to switch this back to false before submitting.

File: go/vt/vtctl/vtctl.go [hunk 2]
```diff
@@ -1941,6 +1953,223 @@ func commandMoveTables(ctx context.Context, wr *wrangler.Wrangler, subFlags *fla
...
+	excludes := subFlags.String("exclude", "", "Tables to exclude (comma-separated) if -all is specified")
+	renameTables := subFlags.Bool("rename_tables", false, "Rename tables instead of dropping them")
+
+	sourceShards := subFlags.String("source_shards", "", "Source shards")
+	targetShards := subFlags.String("target_shards", "", "Target shards")
+	skipSchemaCopy := subFlags.Bool("skip_schema_copy", false, "Skip copying of schema to target shards")
+
+	_ = subFlags.Bool("v2", true, "")
+
+	_ = dryRun //TODO: add dry run functionality
```
Comment (sougou) [inline go/vt/vtctl/vtctl.go:1992]: I saw some code for dry run.  Do you just have to pass the flag in?

**Review by sougou — APPROVED** [review]
I eyeballed the changes. They look good. Couple comments/questions. If they're addressed, this can be merged.

File: go/vt/vtctl/vtctl.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/vt/vtctl/vtctl.go:1992]: Most of the original dry run code will still apply here, but wanted to review and update it if necessary for the new wrappers as a separate PR that should be ready this week.

**Pull request merged**